packages/testnet-faucets/src/ts/handler.ts

```diff
@@ -17,62 +19,55 @@ import { ZRXRequestQueue } from './zrx_request_queue';
 // tslint:disable-next-line:ordered-imports
 import * as Web3 from 'web3';
 
+interface RequestQueueByNetworkId {
+    [networkId: string]: RequestQueue;
+}
+
+const DEFAULT_NETWORK_ID = 42; // kovan
+
 export class Handler {
-    private _etherRequestQueue: EtherRequestQueue;
-    private _zrxRequestQueue: ZRXRequestQueue;
-    private _web3: Web3;
+    private _etherRequestQueueByNetworkId: RequestQueueByNetworkId = {};
+    private _zrxRequestQueueByNetworkId: RequestQueueByNetworkId = {};
     constructor() {
-        // Setup provider engine to talk with RPC node
-        const providerObj = this._createProviderEngine(configs.RPC_URL);
-        this._web3 = new Web3(providerObj);
-
-        this._etherRequestQueue = new EtherRequestQueue(this._web3);
-        this._zrxRequestQueue = new ZRXRequestQueue(this._web3);
+        _.forIn(rpcUrls, (rpcUrl: string, networkId: string) => {
+            const providerObj = this._createProviderEngine(rpcUrl);
+            const web3 = new Web3(providerObj);
+            this._etherRequestQueueByNetworkId[networkId] = new EtherRequestQueue(web3);
+            this._zrxRequestQueueByNetworkId[networkId] = new ZRXRequestQueue(web3, +networkId);
+        });
     }
     public dispenseEther(req: express.Request, res: express.Response) {
-        const recipientAddress = req.params.recipient;
-        if (_.isUndefined(recipientAddress) || !this._isValidEthereumAddress(recipientAddress)) {
-            res.status(400).send('INVALID_REQUEST');
-            return;
-        }
-        const lowerCaseRecipientAddress = recipientAddress.toLowerCase();
-        const didAddToQueue = this._etherRequestQueue.add(lowerCaseRecipientAddress);
-        if (!didAddToQueue) {
-            res.status(503).send('QUEUE_IS_FULL');
-            return;
-        }
-        utils.consoleLog(`Added ${lowerCaseRecipientAddress} to the ETH queue`);
-        res.status(200).end();
+        this._dispense(req, res, this._etherRequestQueueByNetworkId, 'ETH');
     }
     public dispenseZRX(req: express.Request, res: express.Response) {
+        this._dispense(req, res, this._zrxRequestQueueByNetworkId, 'ZRX');
+    }
+    private _dispense(
+        req: express.Request,
+        res: express.Response,
+        requestQueueByNetworkId: RequestQueueByNetworkId,
+        assetSymbol: string,
+    ) {
         const recipientAddress = req.params.recipient;
         if (_.isUndefined(recipientAddress) || !this._isValidEthereumAddress(recipientAddress)) {
-            res.status(400).send('INVALID_REQUEST');
+            res.status(400).send('INVALID_RECIPIENT_ADDRESS');
+            return;
+        }
+        const networkId = _.get(req.query, 'networkId', DEFAULT_NETWORK_ID);
```

Comment:
What if this is `undefined`? It just makes the next `_.get` return undefined? If so, this is relying on a behavior of `_.get` that isn't explicitly mentioned in their docs. Maybe we should explicitly check for `undefined`?

Reply:
It's my understanding that it is undefined wit will just return DEFAULT_NETWORK_ID which is defined as 42 in this case

Reply:
because the next `_.get()` call does not have a default return value it may be `undefined` which we check in the next line before returning `INVALID_NETWORK_ID`

Reply:
Ah yes, you are right!